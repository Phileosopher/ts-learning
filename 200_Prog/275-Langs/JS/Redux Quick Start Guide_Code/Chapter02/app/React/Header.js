import React from "react";
import Link from "react-router-dom/Link";

const Header = () => (
  <ul>
    <li>
      <Link to="/">Home</Link>
    </li>
    <li>
      <Link to="/about">Services</Link>
    </li>
    <li>
      <Link to="/topics">Contact Us</Link>
    </li>
    <li>
      <Link to="/topics">Login</Link>
    </li>
  </ul>
);

export default Header;
