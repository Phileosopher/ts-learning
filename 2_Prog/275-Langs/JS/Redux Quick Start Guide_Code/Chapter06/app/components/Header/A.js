import styled from 'styled-components';
import { Link } from 'react-router-dom';

const A = styled(Link)`
  padding: 5px 15px;
  display: inline-block;
`;

export default A;
