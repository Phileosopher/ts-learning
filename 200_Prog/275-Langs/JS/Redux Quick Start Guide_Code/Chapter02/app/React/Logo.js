import React from "react";
import Link from "react-router-dom/Link";

const Logo = () => (
  <div>
    <Link to="/">FastDoctor</Link>
  </div>
);

export default Logo;
