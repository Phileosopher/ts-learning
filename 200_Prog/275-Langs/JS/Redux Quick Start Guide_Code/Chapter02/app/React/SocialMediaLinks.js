import React from "react";
import Link from "react-router-dom/Link";

const SocialMediaLinks = () => (
  <ul>
    <li>
      <Link to="https://www.facebook.com/jagapati">Facebook</Link>
    </li>
    <li>
      <Link to="https://github.com/sureshHARDIYA">GitHub</Link>
    </li>
    <li>
      <Link to="https://www.researchgate.net/profile/Suresh_Mukhiya2">
        ResearchGate
      </Link>
    </li>
    <li>
      <Link to="https://twitter.com/dr_code_skm">Twitter</Link>
    </li>
  </ul>
);

export default SocialMediaLinks;
