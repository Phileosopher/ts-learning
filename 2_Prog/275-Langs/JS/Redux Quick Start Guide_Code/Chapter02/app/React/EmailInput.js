import React, { Component } from "react";

class EmailInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ""
    };
    this.handleEmailChange = this.handleEmailChange.bind(this);
  }

  handleEmailChange(event) {
    this.setState({ value: event.target.value });
  }

  render() {
    return (
      <div>
        <label>Enter Your Email</label>
        <input
          type="email"
          onChange={this.handleEmailChange}
          value={this.state.value}
          name="email"
        />
      </div>
    );
  }
}

export default EmailInput;
