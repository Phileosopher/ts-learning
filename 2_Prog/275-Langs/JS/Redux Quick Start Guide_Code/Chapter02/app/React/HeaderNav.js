import React from "react";
import Link from "react-router-dom/Link";

import Header from "./Header";
import SocialMediaLinks from "./SocialMediaLinks";
import Logo from "./Logo";

const HeaderNav = () => (
  <div>
    <Logo />
    <Header />
    <SocialMediaLinks />
  </div>
);

export default HeaderNav;
