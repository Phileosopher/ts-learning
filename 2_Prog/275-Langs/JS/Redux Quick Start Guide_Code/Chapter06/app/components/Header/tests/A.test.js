import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import A from '../A';

describe('<A />', () => {
  const id = 'hallo';
  let renderedComponent;
  const InitComponent = (
    <MemoryRouter>
      <A to="/" id={id}>
        test
      </A>
    </MemoryRouter>
  );

  beforeAll(() => {
    renderedComponent = mount(InitComponent);
  });

  it('should match snapshot', () => {
    const rendereredComponent = renderer.create(InitComponent).toJSON();
    expect(rendereredComponent).toMatchSnapshot();
  });

  it('should have a className attribute', () => {
    expect(renderedComponent.find('a').prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    expect(renderedComponent.find('a').prop('id')).toEqual(id);
  });
});
