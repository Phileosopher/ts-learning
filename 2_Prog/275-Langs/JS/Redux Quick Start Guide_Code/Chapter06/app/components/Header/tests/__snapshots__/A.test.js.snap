// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> should match snapshot 1`] = `
.c0 {
  padding: 5px 15px;
  display: inline-block;
}

<a
  className="c0"
  href="/"
  id="hallo"
  onClick={[Function]}
>
  test
</a>
`;
